File: trp/cginfo/cginfo.rb
# Trisul Remote Protocol TRP Demo script
#
# Counter Group Info
#
# Prints information about all supported couner  groups on a trisul instance
#
require 'trisulrp'


USAGE = "Usage:   cginfo.rb   ZMQ_ENDPOINT CGGUID\n" \
        "Example: 1) ruby cginfo.rb ipc:///usr/local/var/lib/trisul/CONTEXT0/run/trp_0 {C51B48D4-7876-479E-B0D9-BD9EFF03CE2E}\n"\
        "         2) ruby cginfoall.rb tcp://localhost:5555 {C51B48D4-7876-479E-B0D9-BD9EFF03CE2E}" 

# usage 
unless ARGV.size==2
  abort USAGE
end

zmq_endpt   = ARGV[0]
target_guid = ARGV[1]
req =TrisulRP::Protocol.mk_request(TRP::Message::Command::COUNTER_GROUP_INFO_REQUEST,
                                    :counter_group => target_guid )

# print a single counter group info
get_response_zmq(zmq_endpt,req) do |resp|
  resp.group_details.each do |group_detail|
     p "Start Time = #{Time.at(group_detail.time_interval.from.tv_sec)}"
     p "End time = #{Time.at(group_detail.time_interval.to.tv_sec)}"
     p "Bucket Size = #{group_detail.bucket_size}"
     p "Name = " + group_detail.name
     p "GUID = " + group_detail.guid
  end
end

File: trp/python/cginfo.py
'''
# CG Info
# Prints all counter groups 
'''

import sys
import datetime
import trp_pb2
import zmq

def usage():
 print "python ", sys.argv[0],"<zmq_endpoint>"
 print "python ", sys.argv[0],"ipc:///usr/local/var/lib/trisul-hub/domain0/hub0/context0/run/trp_0"


if len(sys.argv) != 2:
  usage()
  sys.exit(2)

 
# helper
def get_response(zmq_endpoint,req):
  #zmq send
  context = zmq.Context()
  socket = context.socket(zmq.REQ)
  socket.connect(zmq_endpoint)
  socket.send(req.SerializeToString())

  #zmq receive
  data=socket.recv()
  resp = unwrap_response(data)
  return resp
  socket.close

# helper
def unwrap_response(data):
  resp = trp_pb2.Message()
  resp.ParseFromString(data)
  for x in  resp.DESCRIPTOR.enum_types:
    name = x.values_by_number.get(int(resp.trp_command)).name
  return {
    'COUNTER_GROUP_INFO_RESPONSE':resp.counter_group_info_response
   }.get(name,resp)

#################
# actual code starts here 
# to retrieve all counter groups send an empty COUNTER_GROUP_INFO_REQUEST

req = trp_pb2.Message()
req.trp_command=req.COUNTER_GROUP_INFO_REQUEST
resp = get_response(sys.argv[1],req)

# display the information 
for cg in resp.group_details:
  print cg.guid  + "," + cg.name 


